janusgraph-cassandra/src/main/java/org/janusgraph/diskstorage/cassandra/thrift/thriftpool/CTConnectionFactory.java

```diff
@@ -135,8 +143,15 @@ public boolean validateObject(String key, CTConnection c) {
                           c, curCfg, c.getConfig());
             }
         }
+        
+        boolean isOpen = c.isOpen();
+        try {
+            c.getClient().describe_version();
```

Comment:
Understand you concern, but in this case it is the only way to check the socket connection is to "try" to use it, if the pipe is broken it won't tell you, at least I can't think of other way. 
I'm not talking about a gracefully closing a socket connection which receives their ACKs properly, but something broken or cut by a firewall or something.

as for the connection itself, this method is only called to validate a connection that was previously used and returned to the pool, so before returning it to be re-used, we just make this call, which will fail, so it will remove it from the pool and make a new raw connection.